File: EvoValueManager.Client/src/interfaces/Management.ts
export interface ManagementAssignment {
    characterId: number;
    challengeId: number;
    state: string;
    details?: string | null;
    isClosed: boolean;
}

export interface ManagementDetails {
    state: string;
    details?: string | null;
    isClosed: boolean;
}

export interface AssignChallenge {
    characterId: number;
    challengeId: number;
    stateId: number;
    details?: string | null;
}

export interface UpdateManagement {
    details?: string | null;
    stateId: number;
}

File: EvoValueManager.Server/Models/Entities/Management.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace EvoCharacterManager.Models.Entities
{
    public class Management
    {
        [Column("id")]
        public int ID { get; set; }

        [Column("character_id")]
        public int CharacterId { get; set; }

        [Column("challenge_id")]
        public int ChallengeId { get; set; }

        [Column("state_id")]
        public int StateId { get; set; }

        [Column("details")]
        public string? Details { get; set; }

        public bool IsClosed { get; set; }
    }
}
